Flagsmith.FlagsmithClient/FlagsmithClient.cs

```diff
@@ -0,0 +1,431 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Text;
+using Newtonsoft.Json;
+using FlagsmithEngine.Environment.Models;
+using FlagsmithEngine;
+using FlagsmithEngine.Interfaces;
+using FlagsmithEngine.Identity.Models;
+using FlagsmithEngine.Trait.Models;
+using Microsoft.Extensions.Logging;
+using System.Threading;
+using Flagsmith.Extensions;
+using System.Linq;
+
+namespace Flagsmith
+{   /// <summary>
+    /// A Flagsmith client.
+    /// Provides an interface for interacting with the Flagsmith http API.
+    /// </summary>
+    /// <exception cref="FlagsmithAPIError">
+    /// Thrown when error occurs during any http request to Flagsmith api.Not applicable for polling or ananlytics.
+    /// </exception>
+    /// <exception cref="FlagsmithClientError">
+    /// A general exception with a error message. Example: Feature not found, etc.
+    /// </exception>
+    public class FlagsmithClient
```

Comment:
As discussed on the other PR, this class should be trimmed right down to only expose the methods available on the python client.

Reply:
updated the access modifiers in `flagsmithclient `, and set methods as public only which are needed.

Reply:
I'm confused why we still have `GetTraits`, `GetTrait`, methods etc. ?

Reply:
i thought to remain existing function , will be removing them.